src/server/context.ts

```diff
@@ -180,8 +197,9 @@ export class ServerContext {
         throw new TypeError("Page is not a child of the basepath.");
       }
       const path = url.substring(baseUrl.length).substring("routes".length);
-      const baseRoute = path.substring(1, path.length - extname(path).length);
-      const name = baseRoute.replace("/", "-");
+      let baseRoute = path.substring(1, path.length - extname(path).length);
+      const name = (basePath + baseRoute).replace(/\//g, "-");
+      baseRoute = basePath + baseRoute;
```

Comment:
A bit unrelated: I feel like the whole code in this file is a bit difficult to follow. Hoping to refactor it sometime in the future.